### PR TITLE
Python functions for Workshop 2023

### PR DESCRIPTION
**Pull request by mbaudin47**
This PR creates new content for the Workshop 2023. 

The main content is:
- `Python functions - Overview.ipynb`: an overview of Python functions
- Exercises:
  - `Fonction Python.ipynb`: exercises on Python functions
  - `test_n_cpus.py`: experiment with the `n_cpus` option of the `PythonFunction` class
  - `wrapper-python-benchmark.py`: a benchmark of the `PythonFunction` and `SymbolicFunction` classes

Other Notebooks are provided to the students:
- `Coupling Tools.ipynb`
- `Fonction symbolique.ipynb`
- `Fonctions paramétriques.ipynb`

### CODE REVIEW

**Comment by jschueller** [comment]
a rebase is needed

**Comment by mbaudin47** [comment]
@jschueller The rebase was done.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
           cd userday2022/functional_sensitivity && pdflatex -halt-on-error functional_sensitivity.tex && pdflatex functional_sensitivity.tex && cd -
           cd userday2022/userday2022relhi/ && pdflatex -halt-on-error userday2022relhi.tex && pdflatex userday2022relhi.tex && cd -
           cd userday2022/ud2022persalys && pdflatex -halt-on-error UsersDay-2022-PERSALYS.tex && pdflatex UsersDay-2022-PERSALYS.tex && cd -
+          cd developer/python_function && bash build.sh && cd -
```
Comment (jschueller) [inline .github/workflows/build.yml:34]: this should be moved in the for loop at line 18 for the developer dir
for this the main .tex file must have the same name as the subdir (python_function)
then your build.sh can be dropped

Comment (mbaudin47) [reply .github/workflows/build.yml]: Ok, fixed.

**Pull request merged**